### ISSUE TITLE
Proper provider implementation

### ISSUE THREAD

**Issue by JackNoordhuis** [dihydrogen-monoxide/Factions-pp#8]
The first official release (First non-development build) should include a fully working provider system with a working API that other plugins can access.

The API should be straight forward and easy to use, other plugins should be able to register new providers, overwrite existing ones and use an existing providers functions.

# Progress
- [x] Implement provider system
- [x] Implement default providers
- [x] Create API

### Planned default providers
- [x] JSON
- [x] YAML
- [ ] MySQL
- [ ] SQLite3

**Comment by HittmanA** [comment]
I agree @JackNoordhuis ! (Little late on the reply...)

**Comment by TheDiamondYT1** [comment]
Not trying to be a dick but commands in one file? seriously?

**Comment by HittmanA** [comment]
Yes I know... I started this back when I knew next to nothing about PHP. I need to move the commands to separate files.

**Comment by HittmanA** [comment]
Actually that should be what I do next.

**Comment by TheDiamondYT1** [comment]
I started with that on my Factions plugin but it was messy so im rewriting it 😜

**Comment by HittmanA** [comment]
Lol ;P

**Comment by HittmanA** [comment]
Are you still working on this project @JackNoordhuis ?

**Comment by HittmanA** [comment]
How is it looking so far @JackNoordhuis ?

**Comment by JackNoordhuis** [comment]
Busy with other private projects atm @HittmanA

**Comment by keepitfresh2** [comment]
@HittmanA I was junkdude can u add me on discord, I would like to revive this ttvkeepitfresh#5556 or on twitter @ttv_keepitfresh
